csrc/python_frontend/python_bindings.cpp

```diff
@@ -677,48 +677,75 @@ void initNvFuserPythonBindings(PyObject* module) {
+      [](MatmulParams& self) { return self.tile_sizes.tile_name.m; },          \
+      [](MatmulParams& self, int64_t x) { self.tile_sizes.tile_name.m = x; }); \
+  matmul_config.def_property(                                                  \
+      #tile_name "_n",                                                         \
+      [](MatmulParams& self) { return self.tile_sizes.tile_name.n; },          \
+      [](MatmulParams& self, int64_t x) { self.tile_sizes.tile_name.n = x; }); \
+  matmul_config.def_property(                                                  \
+      #tile_name "_k",                                                         \
+      [](MatmulParams& self) { return self.tile_sizes.tile_name.k; },          \
+      [](MatmulParams& self, int64_t x) { self.tile_sizes.tile_name.n = x; });
+  TILEPARAM(cta_tile)
+  TILEPARAM(warp_tile)
+#undef TILEPARAM
+#define CBPARAM(type, name)                                                 \
+  matmul_config.def_property(                                               \
+      #name,                                                                \
+      [](MatmulParams& self) { return self.circular_buffer_options.name; }, \
+      [](MatmulParams& self, type x) {                                      \
+        self.circular_buffer_options.name = x;                              \
       });
-  pointwise_config.def_property(
-      "vectorization_factor",
-      [](PointwiseParams& self) { return self.vectorization_factor; },
-      [](PointwiseParams& self, int64_t vectorization_factor_) {
-        self.vectorization_factor = vectorization_factor_;
+  CBPARAM(bool, circular_buffer_smem_write)
+  CBPARAM(bool, circular_buffer_smem_read)
+  CBPARAM(int, smem_circular_buffer_stage)
+#undef CBPARAM
+  PARAM(matmul_config, MatmulParams, bool, async_gmem_load_operands)
+  PARAM(matmul_config, MatmulParams, bool, rotate_ldmatrix_out_of_main_loop)
+  PARAM(matmul_config, MatmulParams, int, grid_swizzle_factor)
+  PARAM(matmul_config, MatmulParams, bool, use_smem_epilogue)
+  PARAM(matmul_config, MatmulParams, bool, promote_prologue_smem_reuse)
+  PARAM(matmul_config, MatmulParams, int, splitk_factor)
+  py::enum_<MatmulParams::TileRasterizationOrder>(
+      nvfuser, "MatmulTileRasterizationOrder")
+      .value("column_major", MatmulParams::TileRasterizationOrder::ColumnMajor)
+      .value("row_major", MatmulParams::TileRasterizationOrder::RowMajor);
+  matmul_config.def_property(
+      "cta_order",
+      [](MatmulParams& self) { return self.cta_order; },
+      [](MatmulParams& self, MatmulParams::TileRasterizationOrder cta_order_) {
+        self.cta_order = cta_order_;
       });
-  pointwise_config.def_property(
-      "unroll_factor",
-      [](PointwiseParams& self) { return self.unroll_factor; },
-      [](PointwiseParams& self, int64_t unroll_factor_) {
-        self.unroll_factor = unroll_factor_;
-      });
-  pointwise_config.def(
```

Comment:
Does `__repr__` still exist? It would be useful for matmul params too.

Reply:
Yeah, it moved to INITPARAMS macro above. I used it for debugging my script. Works as expected: `print(params)`

Reply:
Ah, I'm blind. 🙈